File: Check-If-String-Is-a-Prefix-of-Array.java
class Solution {
    public boolean isPrefixString(String s, String[] w) {
        String t = "";
        for (int i = 0; i < w.length; i++) 
        {
            t += w[i];
            if (t.equals(s)) 
            {
                return true;
            }
            if (t.length() > s.length()) 
            {
                return false;
            }
        }
        return false;
    }
}

File: Check-If-String-Is-a-Prefix-of-Array.cpp
class Solution 
{
public:
    bool isPrefixString(string s, vector<string>& w) 
    {
        string t = "";
        for (int i = 0; i < w.size(); i++) 
        {
            t += w[i];
            if (t == s) {
                return true;
            }
            if (t.length() > s.length()) {
                return false;
            }
        }
        return false;
    }
};
